sklearn/linear_model/_logistic.py

```diff
@@ -865,6 +894,33 @@ def _logistic_regression_path(X, y, pos_class=None, Cs=10, fit_intercept=True,
         w0 = np.zeros((classes.size, n_features + int(fit_intercept)),
                       order='F', dtype=X.dtype)
 
+    # preconditioning
+    X_pre = X
+    X_scale = None
+    X_offset = None
+    if precondition and solver == 'lbfgs':
+        # FIXME this duplicates come code from _preprocess_data
+        # and should be refactored
+        if sparse.issparse(X):
+            X_mean, X_var = mean_variance_axis(X, axis=0)
+            X_scale = np.sqrt(X_var, X_var)
+            X_scale[X_scale == 0] = 1
+
+            del X_var
+            if fit_intercept:
+                X_offset = -X_mean
+            # can we actually do inplace here?
+            inplace_column_scale(X_pre, 1 / X_scale)
```

Comment:
should this be inplace?

Reply:
we copy if it's not sparse

Reply:
Sorry not sure if I understand your second comment.